### PR TITLE
Added xxxApprox overloads for YUV color conversions in HAL and AlgorithmHint to cvtColor

### PR DESCRIPTION
**Pull request by asmorkalov**
The xxxApprox to implement HAL functions with less bits for arithmetic of FP.

The hint was introduced in #25792 and #25911

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by asmorkalov** [comment]
Perf statistics for Jetson tk1 (ARMv7 NEON) and jetson orin:
[perf-cvtcolor.zip](https://github.com/user-attachments/files/16285196/perf-cvtcolor.zip)
Looks like the HAL makes sense for old v7 chips only.

File: modules/imgproc/src/hal_replacement.hpp [hunk 1]
```diff
@@ -449,6 +449,23 @@ inline int hal_ni_cvtGraytoBGR5x5(const uchar * src_data, size_t src_step, uchar
...
+   @param dst_step destination image step
+   @param width image width
+   @param height image height
+   @param depth image depth (one of CV_8U, CV_16U or CV_32F)
+   @param scn source image channels (3 or 4)
+   @param swapBlue if set to true B and R source channels will be swapped (treat as RGB)
+   @param isCbCr if set to true write output in YCbCr format
+   Convert from BGR, RGB, BGRA or RGBA to YUV or YCbCr.
+ */
+inline int hal_ni_cvtBGRtoYUVApprox(const uchar * src_data, size_t src_step, uchar * dst_data, size_t dst_step, int width, int height, int depth, int scn, bool swapBlue, bool isCbCr) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
```
Comment (vpisarev) [inline modules/imgproc/src/hal_replacement.hpp:466]: why 'Approx' is in the function name? Why not pass algorithm hint as a parameter?

Comment (asmorkalov) [reply modules/imgproc/src/hal_replacement.hpp:466]: The existing API is already used in 3rd party HAL like Carotene, KleidiCV and some other. So we need to itnroduce new API anyway. The solution just follows current HAL design.

**Review by vpisarev — APPROVED** [review]

**Comment by asmorkalov** [comment]
@opencv-alalek Could you update ABI complience checker on BuildBot?

**Pull request merged**